src/main/java/application/MoveSampleStatus.java

```diff
@@ -0,0 +1,59 @@
+package application;
+
+import domain.EventStore;
+import domain.sample.Sample;
+import domain.sample.SampleCode;
+import domain.sample.SampleEvent;
+import java.time.Instant;
+import java.util.SortedSet;
+import java.util.function.Consumer;
+
+/**
+ * <b>short description</b>
+ *
+ * <p>detailed description</p>
+ *
+ * @since <version tag>
+ */
+public class MoveSampleStatus {
+
+  private final EventStore eventStore;
+
+  public MoveSampleStatus(EventStore eventStore) {
+    this.eventStore = eventStore;
+  }
+
+  public void moveSample(String sampleCode, String sampleStatus, String instant) {
+    SampleCode code = SampleCode.fromString(sampleCode);
+    Instant performAt = Instant.parse(instant);
+    // restore the status
+    SortedSet<SampleEvent> sampleEvents = eventStore.findForSample(code);
+    Sample sample = Sample.create(code);
+    sampleEvents.forEach(sample::addEvent);
+    // run the command
+    determineCommand(performAt, sampleStatus).accept(sample);
+    // store events
+    sample.events().forEach(eventStore::store);
+  }
+
+  private Consumer<Sample> determineCommand(Instant performAt, String sampleStatus) {
```

Comment:
NIT: I like the idea of this method, but I think the method names should be past tense, as well. The sample is not sequenced in this function, it has already been sequenced and the status is changed to reflect that.

Reply:
You have a point here. I was not sure how to implement this as commands usually cause the event and do not only record it. 
I chose to use the names I picked because the samples in the tracking system are not sequenced. What sequence() does is up to the sample at that point.
Do you have suggestions on how to change the method names?